src/rust/engine/fs/store/src/remote/reapi.rs

```diff
@@ -225,13 +261,24 @@ impl ByteStoreProvider for Provider {
       max_batch_total_size_bytes == 0 || len < max_batch_total_size_bytes;
 
     let result = if batch_api_allowed_by_local_config && batch_api_allowed_by_server_config {
-      self.store_bytes_source_batch(digest, bytes).await
+      self.store_bytes_batch(digest, bytes).await
     } else {
-      self.store_bytes_source_stream(digest, bytes).await
+      self
+        .store_source_stream(digest, Box::new(Cursor::new(bytes)))
+        .await
     };
     result.map_err(|e| e.to_string())
   }
 
+  async fn store(&self, digest: Digest, source: StoreSource) -> Result<(), String> {
+    // an arbitrary source (e.g. file) might be small enough to write via the batch API, but we
+    // ignore that possibility for now
```

Comment:
Is it worth considering this optiisation? This which would mean pulling the entire (sufficiently-small) `source` into a `Bytes` to be able to call `self.store_bytes_batch`, similar to the `store_bytes` case.

Reply:
It shouldn't be, do the limitations on when FSDB is used. And right now that is the only source of data (we always capture locally and then upload).

Reply:
Hm, just noting that the batch API size limit is 4194304 (4 MiB) which is greater than the FSDB limit of 512 KiB, i.e. the REAPI code would be happy enough to upload some moderate FSDB-sourced-files via the batch API (and, if we supported uploading multiple files in a batch, we could upload up to 8 FSDB-sourced files in one batch).

I will thus leave this comment here, but not do any action with it for now.